internal/graph/track_check_resolver_test.go

```diff
@@ -0,0 +1,154 @@
+package graph
+
+import (
+	"context"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/oklog/ulid/v2"
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	parser "github.com/openfga/language/pkg/go/transformer"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/goleak"
+	"go.uber.org/mock/gomock"
+
+	"github.com/openfga/openfga/pkg/logger"
+	"github.com/openfga/openfga/pkg/storage"
+	"github.com/openfga/openfga/pkg/storage/memory"
+	"github.com/openfga/openfga/pkg/tuple"
+	"github.com/openfga/openfga/pkg/typesystem"
+)
+
+func TestIntegrationTracker(t *testing.T) {
+	t.Cleanup(func() {
+		goleak.VerifyNone(t)
+	})
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	logger := logger.NewNoopLogger()
+	cycleDetectionCheckResolver := NewCycleDetectionCheckResolver()
+	t.Cleanup(cycleDetectionCheckResolver.Close)
+
+	localChecker := NewLocalChecker()
+	t.Cleanup(localChecker.Close)
+
+	trackChecker := NewTrackCheckResolver(
+		WithTrackerContext(ctx),
+		WithTrackerLogger(logger))
+	t.Cleanup(trackChecker.Close)
+
+	cycleDetectionCheckResolver.SetDelegate(trackChecker)
+	trackChecker.SetDelegate(localChecker)
+	localChecker.SetDelegate(cycleDetectionCheckResolver)
+
+	t.Run("tracker_delegates_request", func(t *testing.T) {
```

Comment:
Why do you need this test? The one below it is named exactly the same and it has a better structure than this one.

Reply:
It verifies tracker is included in chain of resolvers and functions. Corrected naming.